src/telliot_feeds/sources/manual_sources/twap_manual_input_source.py

```diff
@@ -0,0 +1,56 @@
+from dataclasses import dataclass
+
+from telliot_feeds.datasource import DataSource
+from telliot_feeds.dtypes.datapoint import datetime_now_utc
+from telliot_feeds.dtypes.datapoint import OptionalDataPoint
+from telliot_feeds.utils.log import get_logger
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class TWAPManualSource(DataSource[float]):
+    """DataSource for a manually-entered TWAP value."""
+
+    def parse_user_val(self) -> float:
+        """Parse user input"""
+
+        print("\nType your TWAP value response and press [ENTER].\n\nFor example, type 1234 or 1234.56")
+
+        num = None
+
+        while num is None:
+            usr_inpt = input()
+
+            try:
+                inpt = float(usr_inpt)
+            except ValueError:
+                print("Invalid input. Enter a decimal value (float).")
+                continue
+            if inpt < 0:
+                print("Invalid input. Number must greater than 0.")
+                continue
+
+            print("\nTWAP value (with 18 decimals of precision) " + f"to be submitted on chain:  {inpt*10**18:.0f}")
```

Comment:
You can just make this one f-string instead of adding them together

Reply:
good catch